lib/stdio/duplex.js

```diff
@@ -31,6 +31,16 @@ export const generatorsToDuplex = (generators, {writableObjectMode, readableObje
 	return duplexStream;
 };
 
+const applyGenerator = async function * ({transform, final}, chunks) {
+	for await (const chunk of chunks) {
+		yield * transform(chunk);
+	}
+
+	if (final !== undefined) {
+		yield * final();
+	}
+};
```

Comment:
This new function is the main change of this PR. This is what allows simplifying the syntax.